AudioKit/Common/Internals/Audio File/AKAudioFile+Async Processing.swift

```diff
@@ -0,0 +1,365 @@
+     background thread. Any UI update should be made using:
+
+     dispatch_async(dispatch_get_main_queue()) {
+     // UI updates...
+     }
+     */
+    public func normalize_Async( baseDir: BaseDirectory = .Temp,
+                                 name: String = "",
+                                 newMaxLevel: Float = 0.0,
+                                 completionCallBack: AKCallback) -> NormalizeProcess {
+
+        let process = NormalizeProcess(sourceFile: self, baseDir: baseDir, name: name, newMaxLevel: newMaxLevel, completionCallBack: completionCallBack)
+        return process
+    }
+    
+    /**
+     Process the current AKAudioFile in background to return an
+     AKAudioFile with another file audio appended, if succeeded
+
+     - Parameters:
+     - file: AKAudioFile to be appended to the current file.
+     - name: the name of the resulting file without its extension (String).
+     - baseDir: where the file will be located, can be set to .Resources,  .Documents or .Temp (Default is .Temp)
+     - completionCallBack : AKCallback that will be triggered as soon as
+     process has been completed or failed.
+
+     - Returns: an AppendProcess Object.
+
+     Notice that completionCallBack will be triggered from a
+     background thread. Any UI update should be made using:
+
+     dispatch_async(dispatch_get_main_queue()) {
+     // UI updates...
+     }
+     */
+    public func append_Async( file: AKAudioFile,
+                              baseDir: BaseDirectory = .Temp,
+                              name: String = "",
+                              newMaxLevel: Float = 0.0,
+                              completionCallBack: AKCallback) -> AppendProcess {
+
+        let process = AppendProcess(sourceFile: self,
+                                    file: file,
+                                    baseDir: baseDir,
+                                    name: name,
+                                    completionCallBack: completionCallBack)
+        return process
+    }
+    
+    
+    /**
+     Process the current AKAudioFile in background to return an
+     AKAudioFile extracted from the current AKAudioFile, if succeeded
+
+     - Parameters:
+     - from: the starting sampleFrame for extraction.
+     - to: the ending sampleFrame for extraction
+     - name: the name of the resulting file without its extension (String).
+     - baseDir: where the file will be located, can be set to .Resources,  .Documents or .Temp (Default is .Temp)
+     - completionCallBack : AKCallback that will be triggered as soon as
+     process has been completed or failed.
+
+     - Returns: an ExtractProcess Object.
+
+     Notice that completionCallBack will be triggered from a
+     background thread. Any UI update should be made using:
+
+     dispatch_async(dispatch_get_main_queue()) {
+     // UI updates...
+     }
+     */
+    public func extract_Async( file: AKAudioFile,
+                               from: Int64 = 0,
```

Comment:
Its not obvious to me that to and from should be samples.  Perhaps fromSample: and toSample: ?